app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt

```diff
@@ -0,0 +1,292 @@
+              sweepAngle = 360f,
+              useCenter = false,
+              style = Stroke(width = strokeWidth.toPx(), cap = StrokeCap.Round))
+
+          // Foreground Arc
+          drawArc(
+              color = PROGRESSION_COLOR_BLUE,
+              startAngle = -90f,
+              sweepAngle = sweepAngle,
+              useCenter = false,
+              style = Stroke(width = strokeWidth.toPx(), cap = StrokeCap.Round))
+        }
+
+        // Percentage Text
+        Text(
+            text = "${(progress * 100).toInt()}%",
+            fontSize = 26.sp,
+            fontWeight = FontWeight.Bold,
+            color = ColorPalette.PRIMARY_TEXT_COLOR)
+      }
+}
+
+@Composable
+fun MetricCard(label: String, value: String) {
+  Box(modifier = Modifier.padding(vertical = 4.dp)) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+        modifier =
+            Modifier.border(width = 1.dp, color = Color.LightGray, shape = RoundedCornerShape(8.dp))
+                .size(120.dp, 50.dp)) {
+          Text(text = label, fontSize = 13.sp, color = ColorPalette.SECONDARY_TEXT_COLOR)
+          Text(
+              text = value,
+              fontSize = 15.sp,
+              fontWeight = FontWeight.Bold,
+              color = ColorPalette.PRIMARY_TEXT_COLOR)
+        }
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun AchievementItem(achievement: Achievement) {
+  Row(
+      modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp, horizontal = 16.dp),
+      verticalAlignment = Alignment.CenterVertically) {
+        Box(
+            modifier = Modifier.size(48.dp).background(Color.LightGray, shape = CircleShape),
+            contentAlignment = Alignment.Center) {
+              Image(
+                  painter = painterResource(id = achievement.icon),
+                  contentDescription = achievement.title,
+                  modifier = Modifier.size(64.dp).clip(CircleShape) // Clip the image to be circular
+                  )
+            }
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Column(modifier = Modifier.weight(1f)) {
+          // Achievement Title
+          Text(
+              text = achievement.title,
+              fontSize = 16.sp,
+              fontWeight = FontWeight.Bold,
+              color = ColorPalette.PRIMARY_TEXT_COLOR)
+
+          // Tags Row
+          FlowRow(
+              horizontalArrangement = Arrangement.spacedBy(4.dp),
+              verticalArrangement = Arrangement.spacedBy(5.dp),
+              modifier = Modifier.padding(top = 4.dp)) {
+                achievement.tags.forEach { tag ->
+                  Text(
+                      text = tag,
+                      fontSize = 12.sp,
+                      color = ColorPalette.SECONDARY_TEXT_COLOR,
+                      modifier =
+                          Modifier.background(Color(0xFFE0E0E0), shape = RoundedCornerShape(4.dp))
```

Comment:
Improvement: use a color in the palette `Color.kt` or define for a good reason a new color in the palette

Reply:
<img width="176" alt="Screenshot 2024-11-09 083331" src="https://github.com/user-attachments/assets/00722b9b-61cf-47cf-96c7-cd003ca6395d">


changed it to `ColorPalette.BORDER_COLOR`